### PR TITLE
refactor: Migrate `blocks/procedures.js` to goog.module syntax

### PR DESCRIPTION
**Pull request by cpcallen**
## The basics

- [X] I branched from develop
- [X] My pull request is against develop
- [X] My code follows the [style guide](
      https://developers.google.com/blockly/guides/modify/web/style-guide)
- [X] My code is presented in the form suggested in the [module
      conversion guide](https://github.com/google/blockly/issues/5026)
- [X] I have run `npm test`.

### Resolves

Part of #5597 

### Proposed Changes

Converts `blocks/procedures.js` to `goog.module` with ES6 `const`/`let`.

### CODE REVIEW

**Comment by cpcallen** [comment]
Converting to draft because I realise I forgot to remove the `@suppress {extraRequire|missingRequire}` directive.  (May or may not be able to remove the rest of the suppressions too.)

File: blocks/procedures.js [hunk 1]
```diff
@@ -414,132 +435,132 @@ Blockly.Blocks['procedures_defnoreturn'] = {
...
   },
-  setStatements_: Blockly.Blocks['procedures_defnoreturn'].setStatements_,
-  updateParams_: Blockly.Blocks['procedures_defnoreturn'].updateParams_,
-  mutationToDom: Blockly.Blocks['procedures_defnoreturn'].mutationToDom,
-  domToMutation: Blockly.Blocks['procedures_defnoreturn'].domToMutation,
-  saveExtraState: Blockly.Blocks['procedures_defnoreturn'].saveExtraState,
-  loadExtraState: Blockly.Blocks['procedures_defnoreturn'].loadExtraState,
-  decompose: Blockly.Blocks['procedures_defnoreturn'].decompose,
-  compose: Blockly.Blocks['procedures_defnoreturn'].compose,
+  setStatements_: Blocks['procedures_defnoreturn'].setStatements_,
```
Comment (rachel-fenichel) [inline blocks/procedures.js:505]: All of these helper functions could actually be defined as file-internal, and then installed on both types of blocks. This structure looks like an attempt to make these functions non-global.

File: blocks/procedures.js [hunk 2]
```diff
@@ -946,68 +967,68 @@ Blockly.Blocks['procedures_callnoreturn'] = {
...
          */
-        const xml = Blockly.utils.xml.createElement('xml');
-        const block = Blockly.utils.xml.createElement('block');
+        const xml = xmlUtils.createElement('xml');
+        const block = xmlUtils.createElement('block');
         block.setAttribute('type', this.defType_);
         const xy = this.getRelativeToSurfaceXY();
-        const x = xy.x + Blockly.SNAP_RADIUS * (this.RTL ? -1 : 1);
-        const y = xy.y + Blockly.SNAP_RADIUS * 2;
+        const x = xy.x + internalConstants.SNAP_RADIUS * (this.RTL ? -1 : 1);
```
Comment (rachel-fenichel) [inline blocks/procedures.js:973]: Flagging a use of `internalConstants` for #5158. Internals are leaking here.

File: blocks/procedures.js (hunk 1 above)
Comment (cpcallen) [reply blocks/procedures.js]: Done, by declaring a common object and then doing `{...common, init: …}`.

File: blocks/procedures.js (hunk 2 above)
Comment (cpcallen) [reply blocks/procedures.js:953]: Noted, but as discussed will leave this as is for now.

**Review by rachel-fenichel — APPROVED** [review]

File: blocks/procedures.js [hunk 3]
```diff
@@ -414,132 +403,157 @@ Blockly.Blocks['procedures_defnoreturn'] = {
...
   getProcedureDef: function() {
     return [this.getFieldValue('NAME'), this.arguments_, true];
   },
-  getVars: Blockly.Blocks['procedures_defnoreturn'].getVars,
-  getVarModels: Blockly.Blocks['procedures_defnoreturn'].getVarModels,
-  renameVarById: Blockly.Blocks['procedures_defnoreturn'].renameVarById,
-  updateVarName: Blockly.Blocks['procedures_defnoreturn'].updateVarName,
-  displayRenamedVar_: Blockly.Blocks['procedures_defnoreturn'].displayRenamedVar_,
-  customContextMenu: Blockly.Blocks['procedures_defnoreturn'].customContextMenu,
-  callType_: 'procedures_callreturn',
```
Comment (cpcallen) [inline blocks/procedures.js:509]: Ooops!  This shouldn't have been moved into the common section; it caused a bug, #5972 (fixed in PR #5974).

**Pull request merged**